### ISSUE TITLE
`google_container_cluster.node_config.oauth_scopes` should be a Set

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-google#303]
_This issue was originally opened by @japgolly as hashicorp/terraform#15743. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Consider this output of `terraform plan`:
```
-/+ google_container_cluster.apps (new resource required)
      ...
      node_config.0.oauth_scopes.#:                          "8" => "8"
      node_config.0.oauth_scopes.0:                          "https://www.googleapis.com/auth/compute" => "https://www.googleapis.com/auth/compute"
      node_config.0.oauth_scopes.1:                          "https://www.googleapis.com/auth/devstorage.read_only" => "https://www.googleapis.com/auth/devstorage.read_only"
      node_config.0.oauth_scopes.2:                          "https://www.googleapis.com/auth/logging.write" => "https://www.googleapis.com/auth/logging.write"
      node_config.0.oauth_scopes.3:                          "https://www.googleapis.com/auth/monitoring.write" => "https://www.googleapis.com/auth/monitoring" (forces new resource)
      node_config.0.oauth_scopes.4:                          "https://www.googleapis.com/auth/service.management.readonly" => "https://www.googleapis.com/auth/monitoring.write" (forces new resource)
      node_config.0.oauth_scopes.5:                          "https://www.googleapis.com/auth/servicecontrol" => "https://www.googleapis.com/auth/service.management.readonly" (forces new resource)
      node_config.0.oauth_scopes.6:                          "https://www.googleapis.com/auth/trace.append" => "https://www.googleapis.com/auth/servicecontrol" (forces new resource)
      node_config.0.oauth_scopes.7:                          "https://www.googleapis.com/auth/monitoring" => "https://www.googleapis.com/auth/trace.append" (forces new resource)
```

It's going to create a new resource because the declared order of values in `node_config.0.oauth_scopes` differs from the observed state. The values are all the same and this update would have no effect. For this attribute (and surely many others in other resources and provides), order doesn't matter.

Providers should be able to mark set-like attributes as being order-insensitive. This would allow Terraform to correctly devise that no update is needed and avoid superfluous updates like the one demonstrated above.

**Comment by apparentlymart** [comment]
(there's some discussion about this in the issue that this was migrated from in the Terraform core repository)

**Comment by danawillow** [comment]
Changed the title of the issue to better reflect its actual needs.

**Comment by danawillow** [comment]
Fixed in #506.
